scripts/ci/cut_ironclaw_release.py

```
@@ -0,0 +1,186 @@
+            raise ReleaseTagError(f"failed to read {tag}: {result.stderr.strip()}")
+        target = json.loads(result.stdout)["object"]
+
+        for _ in range(MAX_ANNOTATED_TAG_DEPTH):
+            object_type = str(target["type"])
+            object_sha = str(target["sha"])
+            if object_type == "commit":
+                return object_sha
+            if object_type != "tag":
+                raise ReleaseTagError(
+                    f"{tag} resolves to unsupported Git object type {object_type!r}"
+                )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)